### ISSUE TITLE
[BUG] PGD with zero epsilon is not the same as Clean accuracy

### ISSUE THREAD

**Issue by cod3r0k** [Harry24k/adversarial-attacks-pytorch#124]
### ✨ Short description of the bug [tl;dr]

Hi everyone. I check torch attack. I can not achieve the same results as I check PGD with `eps=0` as clean accuracy. What's happened? Also, I check `alpha=0` and `eps=0`; again, it is not the same as clean accuracy. It was a big bug I think. Isn't it?

### 💬 Detailed code and results

Hi everyone. I check torch attack. I can not achieve the same results as I check PGD with `eps=0` as clean accuracy. What's happened? Also, I check `alpha=0` and `eps=0`; again, it is not the same as clean accuracy. It was a big bug I think. Isn't it?

**Comment by rikonaka** [comment]
Dear @cod3r0k , I think this is because you did not call `model.eval()` before testing.

**Comment by cod3r0k** [comment]
> Dear @cod3r0k , I think this is because you did not call `model.eval()` before testing if your model is using something like `BatchNorm`. So that is not bug. 🤪

mmm Dear @rikonaka 

I reviewed my code and everything is the same as you said. In the previous cells, the mode was changed to "eval", but I repeated it in this cell to increase your credibility. Also, I changed `step` parameter to zero but the results are not convincing.

![image](https://user-images.githubusercontent.com/36899025/224082763-1924b0b4-8942-463c-9c98-79aa6cab1e76.png)

**Comment by rikonaka** [comment]
Hi @cod3r0k , I test it in my code, but I got the same acc as clean examples.

![test](https://user-images.githubusercontent.com/13602602/224248756-70a8750f-0600-44e0-abb6-fa2e508f8524.png)

You can download it and run for test.

[example.zip](https://github.com/Harry24k/adversarial-attacks-pytorch/files/10939536/example.zip)

**Comment by cod3r0k** [comment]
Dear @rikonaka 
I appreciate your help. For you to be able to see better how my code changes, I also develop it online. 
Could you please check it online and let me know what you think?  https://colab.research.google.com/drive/17ABEI8FgEDi9KqAyRl-Sue_alCDZgIAs?usp=sharing

Thanks

**Comment by rikonaka** [comment]
> Dear @rikonaka I appreciate your help. For you to be able to see better how my code changes, I also develop it online. Could you please check it online and let me know what you think? https://colab.research.google.com/drive/17ABEI8FgEDi9KqAyRl-Sue_alCDZgIAs?usp=sharing
> 
> You can see that it has a bug @ evaluating PGD ![image](https://user-images.githubusercontent.com/36899025/224309497-4dfb1be2-dee4-4c6f-a349-4312496663a4.png)
> 
> Thanks

Hi @cod3r0k , I found the problem in your code.

You used normalize when loading the code causing the sample interval to become -1 to 1, while the input limit of `torchattack` is 0 to 1.

![image](https://user-images.githubusercontent.com/13602602/224345735-a564897f-e21a-4d1d-ac84-a86446d0f706.png)

![image](https://user-images.githubusercontent.com/13602602/224345883-09020a24-e40b-4d4b-bec9-794adf943afc.png)

[My pull](https://github.com/Harry24k/adversarial-attacks-pytorch/pull/119/commits/e287472bc6de567152a9884eb44b9866a4a752c9#diff-fa2477f738d9d95acc71e7336cbbc4dbd0be1e34ad7ff41220cf1285e7cb0b09)
![image](https://user-images.githubusercontent.com/13602602/224346503-b6e9a202-57c6-4c9b-90dc-3ab0a4112d1f.png) is here but still not be merge yet.

So make sure your input is 0 to 1.

**Comment by cod3r0k** [comment]
Dear @rikonaka , Thank you for your great note! Wow! :)

But, this means that in the training phase, we also must normalize our dataset with mean 0 and std 1. Isn't it? It means that we must pay more attention to it. Maybe our comparisons are not comparable with this note .....

**Comment by cod3r0k** [comment]
Also, Is it necessary to normalize our data every time with zero mean?

**Comment by rikonaka** [comment]
Dear @cod3r0k, you can always do normalization, but when you feed the images to `torchattack` attacks, just denormalize them 🤪.

**Comment by Harry24k** [comment]
I have added the function `atk.set_normalization_used`!
I will definitely put some explanation on this function.
Sorry for the lack of explanation and the late response.
